src/exporters/certHelpers.go

```diff
@@ -35,41 +37,66 @@ func secondsToExpiryFromCertAsBase64String(s string) ([]certMetric, error) {
 	return secondsToExpiryFromCertAsBytes(certBytes)
 }
 
+func decodeFromPKCS(certBytes []byte) ([]*pem.Block, error) {
+	var blocks []*pem.Block
+	pfx_blocks, err := pkcs12.ToPEM(certBytes, "")
+	if err != nil {
+		return nil, err
+	}
+	for _ , b := range pfx_blocks {
+		if b.Type == "CERTIFICATE" {
+			blocks = append(blocks, b)
+		}
+	}
+	return blocks, nil
+}
+
+
 func secondsToExpiryFromCertAsBytes(certBytes []byte) ([]certMetric, error) {
 	var metrics []certMetric
 	var blocks []*pem.Block
+	var last_err error
 
 	// Export the first certificates in the certificate chain
 	block, rest := pem.Decode(certBytes)
-	if block == nil {
-		return metrics, fmt.Errorf("Failed to parse as a pem")
-	}
-	blocks = append(blocks, block)
-
-	// Export the remaining certificates in the certificate chain
-	for len(rest) != 0 {
-		block, rest = pem.Decode(rest)
-		if block == nil {
-			return metrics, fmt.Errorf("Failed to parse intermediate as a pem")
+	if block != nil {
+		blocks = append(blocks, block)
+		// Export the remaining certificates in the certificate chain
+		for len(rest) != 0 {
+			block, rest = pem.Decode(rest)
+			if block == nil {
+				return metrics, fmt.Errorf("Failed to parse intermediate as a pem")
+			}
+			if block.Type == "CERTIFICATE" {
+				blocks = append(blocks, block)
+			}
 		}
-		if block.Type == "CERTIFICATE" {
-			blocks = append(blocks, block)
+	} else {
+		// Parse as PKCS ?
+		pfx_blocks, err := decodeFromPKCS(certBytes)
+		if err != nil {
+			return metrics, fmt.Errorf("Failed to parse certificate")
 		}
+		blocks = pfx_blocks	
 	}
 
 	for _, block := range blocks {
 		cert, err := x509.ParseCertificate(block.Bytes)
-		if err != nil {
-			return metrics, err
-		}
+		if err == nil {
```

Comment:
why are we not immediately returning this error? it seems like with this change we can drop an error if `len(metrics) > 0`

Reply:
Good question, i capture the last error... and return it at the end in case no metrics are collection.  I locally tested with different production ready certificates (we got from valid CA authorities)... i have discovered that some intermediate 'block' ones are not parsed because some optional NULL properties)... the whole certificate was not being parsed.

Reply:
I'm concerned that this change would hide parsing errors and operators would lose visibility on their certs without realizing it. Without seeing the error it's tough to give advice on how to proceed here.

One option would be to change every place we call this method to publish metrics even if an error is returned. So we would log/metric the error as well as publish metrics on whatever we did happen to parse correctly.